backend/api/Helpers/Routes/Constraints/PidConstraint.cs

```diff
@@ -0,0 +1,51 @@
+using System;
+using System.Globalization;
+using System.Text.RegularExpressions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Routing;
+
+namespace Pims.Api.Helpers.Routes.Constraints
+{
+    /// <summary>
+    /// PidConstraint class, provides a way to add a formatted PID (i.e. 123-123-123) constraint to a route template.
+    /// </summary>
+    public class PidConstraint : IRouteConstraint
```

Comment:
Created this separately because of the apparent security issue with regex requires a timeout.